counterpartylib/lib/blocks.py

```diff
@@ -179,10 +182,11 @@ def parse_block(db, block_index, block_time, previous_ledger_hash=None,
     cursor.close()
 
     # Calculate consensus hashes.
-    new_txlist_hash = check.consensus_hash(db, 'txlist_hash', previous_txlist_hash, txlist)
-    new_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
+    new_txlist_hash, found_txlist_hash = check.consensus_hash(db, 'txlist_hash', previous_txlist_hash, txlist)
+    new_ledger_hash, found_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
+    new_messages_hash, found_messages_hash = check.consensus_hash(db, 'messages_hash', previous_messages_hash, database.BLOCK_MESSAGES)
 
-    return new_ledger_hash, new_txlist_hash
+    return new_ledger_hash, new_txlist_hash, new_messages_hash, found_messages_hash
```

Comment:
Not necessary for the other hashes, as we will throw an exception if they don't match up. With `messages_hash`, _currently_ we are simply displaying that it's a mismatch, for the reasons I outlined in the PR description.